app/components/role_type_component.html.erb

```diff
@@ -0,0 +1,15 @@
+<div class="mt-6 border-t border-gray-200 pt-6 space-y-8">
+  <div>
+    <h2 class="text-sm font-medium text-gray-500">Interested in roles</h2>
+    <ul role="list" class="mt-3 space-y-3">
+      <% role_types.each do |role| %>
+        <% if developer.send("#{role}?") %>
```

Comment:
Is this equivalent?

```suggestion
      <% role_types.select { |rt| developer.public_send "#{role}?" }.each do |role| %>
```

Alternatively, you could add a `developer_role_types` method to the `RoleTypeComponent` to DRY up the logic e.g.

```ruby
  def developer_role_types
    role_types.select do |rt| 
      developer.public_send "#{role}?"
    end
  end
```

thus simplifying the view logic to `<% developer_role_types.each do |role| %>` and the component's render logic to:

```ruby
  def render?
    developer_role_types.any?
  end
```

Reply:
Ohh, the methods added to `RoleTypeComponent` are a great call! I just applied that suggestion.